File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r:
            midpoint = ((r - l) // 2) + l
            num = nums[midpoint]
            if num < target:
                l = midpoint + 1
            elif num > target:
                r = midpoint - 1
            else:
                return midpoint
        return -1

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int zeroCount = 0;
        int totalProduct = 1;
        int[] result = new int[nums.length];
        for (Integer num : nums) {
            if (num == 0) zeroCount++;
            else totalProduct = totalProduct * num;
        }
        for (int i = 0; i < nums.length; i++) {
            if (zeroCount == 1) {
                if (nums[i] != 0) result[i] = 0;
                else result[i] = totalProduct;
            } else if (zeroCount > 1) {
                result[i] = 0;
            } else {
                result[i] = totalProduct / nums[i];
            }
        }
        return result;
    }
}  
